src/js/player.js

```diff
@@ -4366,6 +4368,8 @@ class Player extends Component {
       return this.audioOnlyMode_;
     }
 
+    this.audioOnlyMode_ = value;
+
```

Comment:
Moved this here because 'this.audioOnlyMode_' was only getting updated after the promise was resolved and so outdated values were retuned when audioOnlyMode() was called in the audioPosterMode method